### PR TITLE
Chore(TDI-48532) Automated malformed encoding fix

### PR DESCRIPTION
**Pull request by fdesiles**
https://jira.talendforge.org/browse/TDI-48532

Updating post login actions to always check for m2 corruption and clean the possible associated items.

### CODE REVIEW

**Review by acatoire — APPROVED** [review]
LGTM

**Comment by fdesiles** [comment]
> LGTM

Updated PR to avoid grep on M2 when no problem exists (adds many minutes to builds).
Now the clean will be done when a Malformed Encoding is detected in the logs. So further builds won't be blocked.

**Review by acatoire — CHANGES_REQUESTED** [review]
Why are you not using the function like in  cloud-components PR?
https://github.com/Talend/cloud-components/blob/e855d4f543feda55ee075c7681fa85bb376130dc/Jenkinsfile#L462-L463

**Comment by fdesiles** [comment]
> Why are you not using the function like in cloud-components PR? https://github.com/Talend/cloud-components/blob/e855d4f543feda55ee075c7681fa85bb376130dc/Jenkinsfile#L462-L463

I do, please recheck :)

**Review by acatoire — APPROVED** [review]
LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -413,3 +431,19 @@ pipeline {
         }
     }
 }
+
+//TODO: https://jira.talendforge.org/browse/TDI-48913 Centralize script for Jenkins M2 Corruption clean
+private void CleanM2Corruption(logContent) {
+//Clean M2 corruptions - TDI-48532
+    if (logContent.contains("Malformed \\uxxxx encoding")) {
+        echo 'Malformed encoding detected: Cleaning M2 corruptions'
```
Comment (fdesiles) [inline Jenkinsfile:439]:
```suggestion
    //Clean M2 corruptions - TDI-48532
    echo 'Checking for Malformed encoding error'
    if (logContent.contains("Malformed \\uxxxx encoding")) {
        echo 'Malformed encoding detected: Cleaning M2 corruptions'
```

**Review by acatoire — APPROVED** [review]
LGTM

**Pull request merged**